hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/StakePeriodManager.java

```diff
@@ -0,0 +1,89 @@
+package com.hedera.services.ledger.accounts.staking;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hedera.services.context.TransactionContext;
+import com.hedera.services.state.merkle.MerkleNetworkContext;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.function.Supplier;
+
+@Singleton
+public class StakePeriodManager {
+	private final TransactionContext txnCtx;
+	private final Supplier<MerkleNetworkContext> networkCtx;
+	private long currentStakePeriod;
+	private long prevConsensusSecs;
+
+	public static final ZoneId zoneUTC = ZoneId.of("UTC");
+
+	@Inject
+	public StakePeriodManager(final TransactionContext txnCtx,
+			final Supplier<MerkleNetworkContext> networkCtx) {
+		this.txnCtx = txnCtx;
+		this.networkCtx = networkCtx;
+	}
+
+	public long currentStakePeriod() {
+		final var currentConsensusSecs = txnCtx.consensusTime().getEpochSecond();
+
+		if (prevConsensusSecs != currentConsensusSecs) {
+			prevConsensusSecs = currentConsensusSecs;
+			currentStakePeriod = LocalDate.ofInstant(txnCtx.consensusTime(), zoneUTC).toEpochDay();
+		}
+		return currentStakePeriod;
+	}
+
+	public long latestRewardableStakePeriodStart() {
```

Comment:
I think given how we use this method now, a more accurate name is `firstNonRewardableStakePeriod()`.

That is, it returns a `stakePeriod` such that iff your `stakePeriodStart < startPeriod`, you are eligible for a reward.